src/Generator.cpp

```diff
@@ -31,291 +38,657 @@ Generator(const Parameters& params):
 {
 }
 
-void Generator::simulate() {
-  unsigned nevents = parameters_.general().events;
-  bool debug = parameters_.general().debug;
 
-  // some initializations
-  double energygen=0.;
-  double energygenincells=0.;
-  double energyrec=0.;
+Generator::~Generator(){
+}
 
 
+std::array<std::array<double, NB_SI_THICKNESS>, NB_LAYERS> Generator::readCalibration(const std::string& filename) {
 
-  TStopwatch t;
-  t.Start();   
+  std::array<std::array<double, NB_SI_THICKNESS>, NB_LAYERS> calib;
 
-  if (parameters_.geometry().type!=Parameters::Geometry::Type::External) {
-    geometry_.constructFromParameters(parameters_.general().debug); 
-  } else {
-    geometry_.constructFromJson(parameters_.general().debug);
+  std::ifstream my_calib_file(filename);
+  std::stringstream ss;
+  std::string line, temp;
+
+  int col_counter = 0;
+  int line_counter = 0;
+
+  int dim_column = 1 + NB_SI_THICKNESS * 2;
+
+  if (my_calib_file) {
+
+    int layer_;
+    double mip[NB_SI_THICKNESS];
+    double noise[NB_SI_THICKNESS];
+    double sampl, MIP;
+
+    while(std::getline(my_calib_file, line)) {
+
+      // Remove comment lines (beginning with #)
+      if (line.find("#") != std::string::npos) {
```

Comment:
This is not doing what is written in the comment. It is discarding any line containing `#`.